Tests/Marketplace/search_and_install_packs.py

```diff
@@ -124,102 +108,51 @@ def create_dependencies_data_structure(response_data: dict, dependants_ids: list
-        pack_data (dict): Contains the pack ID and version.
+        pack_id (str): ID of the pack to get dependencies for.
         lock (Lock): A lock object.
+
     Returns:
-        (list) The pack's dependencies.
+        dict: API response data for the /search/dependencies endpoint.
     """
     global SUCCESS_FLAG
-    pack_id = pack_data['id']
-    logging.debug(f'Getting dependencies for pack {pack_id}')
+
+    api_endpoint = "/contentpacks/marketplace/search/dependencies"
+    body = [{"id": pack_id}]  # 'version' key can be specified alongside "id". Not including it will fetch the latest version.
+
+    logging.debug(f"Fetching dependencies for pack '{pack_id}'.\n"
+                  f"Sending POST request to {api_endpoint} with body: {json.dumps(body)}")
+
     try:
-        response_data, status_code, _ = demisto_client.generic_request_func(
+        response_data, _, _ = demisto_client.generic_request_func(
             client,
-            path='/contentpacks/marketplace/search/dependencies',
+            path=api_endpoint,
             method='POST',
-            body=[pack_data],
+            body=body,
             accept='application/json',
             _request_timeout=None,
-            response_type='object'
+            response_type='object',
         )
-        if 200 <= status_code < 300:
-            dependencies_data: list = []
-            dependants_ids = [pack_id]
-            response_data = response_data.get('dependencies', [])
-            create_dependencies_data_structure(response_data, dependants_ids, dependencies_data, dependants_ids)
-            if dependencies_data:
-                dependencies_str = ', '.join([dep['id'] for dep in dependencies_data])
-                logging.debug(f'Found the following dependencies for pack {pack_id}: {dependencies_str}')
-            return dependencies_data
-        if status_code == 400:
-            logging.error(f'Unable to find dependencies for {pack_id}.')
-            return []
-        msg = response_data.get('message', '')
-        raise Exception(f'status code {status_code}\n{msg}\n')
+
+        return response_data
+
     except ApiException as api_ex:
         with lock:
             SUCCESS_FLAG = False
-        logging.exception(f"The request to get pack {pack_id} dependencies has failed, Got {api_ex.status} from server, "
-                          f"message:{api_ex.body}, headers:{api_ex.headers}")
+        logging.exception(f"API request to fetch dependencies of pack '{pack_id}' has failed.\n"
+                          f"Response code '{api_ex.status}'\nResponse: '{api_ex.body}'\nResponse Headers: '{api_ex.headers}'")
+
     except Exception as ex:
         with lock:
             SUCCESS_FLAG = False
-        logging.exception(f"The request to get pack {pack_id} dependencies has failed. {ex}.")
+        logging.exception(f"API call to fetch dependencies of '{pack_id}' has failed.\nError: {ex}.")
 
-
-def search_pack(client: demisto_client,
```

Comment:
Make sure that we are not using this function in upload flow/private build/ other places in content

Reply:
Already tested, it's not.